[WIP] GitHub developing branch

jam-world opened the pull request:
Developing the github driver.

TASKS:
  - [x] list repo

jam-world committed:
[ADD] package-lock.json

New feature for the npm package management

jam-world committed:
[UPDATE] test.config.js.example

We need username and test repo for the general test.

jam-world committed:
[UPDATE] jsonRequest.js

Throw exception when the response status code is not success.

jam-world committed:
[FIX] github-api.spec.js

Now we can pass all the github and dropbox test

geyang commented on src/github/github-api.spec.js:
```diff
@@ -106,21 +110,24 @@ describe("repository operations", function () {
       done()
+    }).catch(error => {
+      console.log(error);
+      fail()
```
:joy: you can just remove this catch because it is not needed for createFile.

geyang commented on src/github/github-api.spec.js:
```diff
@@ -129,19 +136,25 @@ describe("repository operations", function () {
       done()
+    }).catch(error => {
+      console.log(error);
+      fail()
```
you should
1. use `throw Error` in this case
2. only call the `done` callback in final (not sure if this is correct b/c forgot the logic of `fetch api`.

geyang commented on test.config.js.example:
```diff
@@ -4,9 +4,15 @@ export const dropboxAccessToken = "your access token";
 
 export const githubClientId = "you app id";
 export const githubClientSecret = "you client secret";
+export const githubClientUsername = "you github username";
```
change to test client username for consistency, move to bottom (L12).

geyang commented on src/github/jsonRequest.js:
```diff
@@ -18,5 +18,10 @@ export default function jsonRequest(access_token, token_type, method = "get", ur
+              throw Error(response.statusText);
+          }
+          return response
+      }).then(res => no_parse ? res : res.json())
```
I think a proper way to handle this would be to use `promise.reject` but I'm not quite sure.

Also besides, `response.ok` is false whenever (if I recall correctly) the status code is anything but `20x`. So I would probably leave this as it was for now and come back for more sophisticated handling later.

geyang replied:
still need to remove, move into `delete` test.

jam-world committed:
[FIX] github-api.spec.js

Remove the wrong fail() usage. Fix the delete file test, so that it will
fail on the wrong response status.

jam-world committed:
[UPDATE] jsonRequest.js

do not handle the wrong response status for now.

geyang commented on src/github/github-api.spec.js:
```diff
@@ -1,7 +1,11 @@
+    githubClientSecret,
+    githubAccessToken as accessToken,
+   githubClientUsername as username,
+   testRepoName} from "../../test.config.js";
```
need to lint.

geyang commented on src/github/github-api.spec.js:
```diff
@@ -1,7 +1,11 @@
+import {githubClientId as clientId,
+    githubClientSecret,
+    githubAccessToken as accessToken,
+   githubClientUsername as username,
```
change to `testUsername`.

jam-world committed:
[REFORMAT] github-api.spec.jso

jam-world committed:
[REFORMAT] jsonRequest.js

jam-world committed:
Revert "[UPDATE] jsonRequest.js"

This reverts commit 82f00cdac0d4d3e242936e2772236600c0a3b31e.

jam-world committed:
Revert "[REFORMAT] github-api.spec.jso"

This reverts commit e56fa1e538e592e6736f7bae6c72ca4b6c9fd043.

jam-world committed:
Revert "Revert "[UPDATE] jsonRequest.js""

This reverts commit e55e0f6baf59fd7604d4d9b45b744c26f7b07abf.

jam-world committed:
Revert "Revert "[REFORMAT] github-api.spec.jso""

This reverts commit dba4b5daedb4b6969b2419212bff186d9979a2ce.

jam-world committed:
Revert "[REFORMAT] jsonRequest.js"

This reverts commit 1023306374d0a3cf12c22e9822c2c38a3460a963.

jam-world committed:
[REMOVE] comment

jam-world committed:
Revert "[REFORMAT] github-api.spec.jso"

This reverts commit e56fa1e538e592e6736f7bae6c72ca4b6c9fd043.

jam-world committed:
[ADD] list repo method

Now we can list the github repo for a given user

jam-world committed:
[ADD] list file method

Now we can list file from given path in a given repo

jam-world committed:
[UPDATE] private test config template

config for test private repo

jam-world committed:
[ADD] test private repo

New test for private repo

jam-world commented:
Ready to merge @episodeyang

geyang commented on test.config.js.example:
```diff
@@ -16,3 +15,6 @@ export const githubAccessToken = "your access token";
  * Make Sure the test repo exists under the given username
  */
 export const testRepoName = "repo name"
+export const testGithubClientUsername = "your github username";
```
should just be `testGithubUsername`

geyang commented on src/github/github-api.spec.js:
```diff
@@ -4,8 +4,10 @@ import GitHubApi from "./github-api";
+        testRepoName,
+        testGithubClientUsername as testUsername,
+        testPublickRepo,
+        testPrivateRepo} from "../../test.config.js";
```
Need to add this to text.config.example as well: ) use these values in that example since they are not sensitive info.

geyang commented:
LGTM

jam-world committed:
[REFACTOR] testGithubClientUseranme to testGithubUsername
